### DISCUSSION TITLE
Expanding JSON information using Dart SDK

### DISCUSSION THREAD

**Discussion by JustJerem** [pocketbase/pocketbase#2418] [Q&A] [answered]
Hello,

I would like more details on how to expand the information of a JSON using the Dart SDK.
I am currently working on a project where I am using the PocketBase Dart SDK to retrieve data from a collection of users. I am trying to expand the "favorites" field in the JSON response, but I have been unsuccessful so far. I have tried using the "expand" parameter in the SDK's getFirstListItem() method, but the "expand" field still appears empty in the response.

I am trying to reproduce this call without success:
```
// or fetch only the first record that matches the specified filter
final record = await pb.collection('users').getFirstListItem(
  'someField="test"',
  expand: 'relField1,relField2.subRelField',
);
```

My version :
```
    var filter = 'id = "$userId"';
    var expand = 'favorites';
    final value = await PocketBaseConnector().getCollectionObjects(
      PocketBaseCollection.users,
      filter,
      expand,
    );
```

```
  Future<List<RecordModel>> getCollectionObjects(
    String collectionId,
    String filter,
    String expand
  ) async {
    var response = await _pocketBase.collection(collectionId).getFirstListItem(filter, expand: expand);
    return response.expand["favorites"] ?? List.empty();
  }
 ```

I am getting this response, but expand is still empty:

```
{
  "id": "ob44o2y6abal2z3",
  "created": "2023-04-29 21:58:40.475Z",
  "updated": "2023-05-03 15:14:50.627Z",
  "collectionId": "_pb_users_auth_",
  "collectionName": "users",
  "expand": {},
  "avatar": "",
  "email": "julien@gmail.com",
  "emailVisibility": false,
  "favorites": [
    "sejny5uaiopnw3b",
    "amjq2h9nlpnm9r3"
  ],
  "name": "",
  "username": "users38122",
  "verified": true
}
```

I would appreciate any help or insights on how to properly expand the "favorites" field in the JSON response using the Dart SDK.
Thank you! 🙏

**Comment by ganigeorgiev** [comment, marked as answer]
Make sure that your client request satisfies the View API rule of the collection related to the `favorites` field.

If that's not the issue, please export your collections configuration (_Admin UI > Settings > Export collections_) so that I can try to reproduce it locally.

_Side-note: if you want to perform a single record retrieve action based on the `id` field, it will be more performant to use directly the `getOne()` method._

**Reply by JustJerem** [reply]
Thank you! 🙏

It makes perfect sense when you think about it. But at the time, I was stuck on my `users`, I did not think to update the rules in `favorites`!

It's fixed now. Thanks for the quick reply.
